src/state/opamRepositoryState.ml

```diff
@@ -111,6 +111,41 @@ let load_opams_from_dir repo_name repo_root =
     (fun () -> aux OpamPackage.Map.empty (OpamRepositoryPath.packages_dir repo_root))
     ~finally:OpamConsole.clear_status
 
+let load_opams_incremental repo_name repo_root changed_files rt =
+  let existing_opams =
+    try OpamRepositoryName.Map.find repo_name rt.repo_opams
+    with Not_found -> OpamPackage.Map.empty
+  in
+  List.fold_left (fun acc file ->
+      let packages_dir = OpamRepositoryPath.packages_dir repo_root in
```

Comment:
this line does not depend on the parameters of the fold so it can be moved outside of it to avoid repeating it for nothing